File: libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKNodeLogger.kt
package com.breezsdk

import breez_sdk.LogMessage
import breez_sdk.Logger
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezSDKNodeLogger(private val emitter: RCTDeviceEventEmitter): Logger {
    companion object {
        var emitterName = "breezSdkNodeLog"
    }

    override fun log(logMessage: LogMessage) {
        emitter.emit(emitterName, readableMapOf(logMessage))
    }
}

File: libs/sdk-react-native/ios/BreezSDKNodeLogger.swift
import Foundation
import BreezSDK

class BreezSDKNodeLogger: NSObject, Logger {
    var emitter: RCTEventEmitter
    
    static let emitterName: String = "breezSdkNodeLog"
    
    init(emitter: RCTEventEmitter) {
        self.emitter = emitter
    }
    
    func log(logMessage: LogMessage) {
        self.emitter.sendEvent(withName: BreezSDKNodeLogger.emitterName, 
                               body: BreezSDKMapper.dictionaryOf(logMessage: logMessage))
    }
}
